snapshots/devmapper/dmsetup/dmsetup.go

```diff
@@ -345,6 +353,39 @@ func BlockDeviceSize(path string) (int64, error) {
 	return size, nil
 }
 
+// DiscardBlocks discards all blocks for the given thin device
+//  ported from https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/pkg/devicemapper/devmapper.go#L416
+func DiscardBlocks(deviceName string) error {
+	inUse, err := isInUse(deviceName)
+	if err != nil {
+		return err
+	}
+	if inUse {
```

Comment:
If we don't do this check, will `BLKDISCARD` syscall return us the same busy err code?

Reply:
No, it will discard blocks and break whatever is using the device.

E.g. using the `blkdiscard` on a running container:
```
$ sudo ctr  run --snapshotter devmapper --rm --tty --net-host docker.io/library/busybox:latest busybox-test
/ #
```
```
$ sudo dmsetup info devpool-snap-13
Name:              devpool-snap-13
State:             ACTIVE
Read Ahead:        256
Tables present:    LIVE
Open count:        1
Event number:      0
Major, minor:      253, 1
Number of targets: 1
$ sudo blkdiscard /dev/mapper/devpool-snap-13
```
```
/ # ls
sh: ls: not found
```